live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java

```diff
@@ -135,6 +114,65 @@ private void writeProperty(JsonGenerator jgen, FieldType fieldType, String name,
         }
     }
 
+    private boolean writeAnnotation(JsonGenerator jgen, String name) throws IOException
+    {
+        boolean hasAnnotation = true;
+        switch (name)
+        {
+        case CREATED_AT:
+            jgen.writeObjectField("annotation", "dateCreated");
+            break;
+        case MODIFIED_AT:
+            jgen.writeObjectField("annotation", "dateModified");
+            break;
+        case ASPECTS_NAMES:
+            jgen.writeObjectField("annotation", "aspects");
+            break;
+        case NAME:
+            jgen.writeObjectField("annotation", "name");
+            break;
+        case TYPE:
+            jgen.writeObjectField("annotation", "type");
+            break;
+        case CREATED_BY:
+            jgen.writeObjectField("annotation", "createdBy");
+            break;
+        case MODIFIED_BY:
+            jgen.writeObjectField("annotation", "modifiedBy");
+            break;
+        default:
+            hasAnnotation = false;
+            break;
+        }
+        return hasAnnotation;
+    }
+
+    private void writeType(JsonGenerator jgen, Object value) throws IOException
+    {
+        if (value instanceof FileMetadata)
+        {
+            return;
+        }
+        String type;
+        if (value instanceof Boolean)
+        {
+            type = "boolean";
+        }
+        else if (value instanceof Integer)
+        {
+            type = "integer";
+        }
+        else if (value instanceof Float)
+        {
+            type = "float";
+        }
+        else
```

Comment:
I think you should also cope with collections. You could refactor this to extract a function:
```
    private String selectTypeByValue(Object value)
    {
        if (value instanceof Boolean)
            ...
        else if (value instanceof Collection)
        {
            Collection collection = (Collection) value;
            if (collection.isEmpty())
            {
                // This should not happen because empty collections should already be filtered out.
                throw new ...;
            }
            return selectTypeByValue(collection.stream().findAny());
        }
        // Default to string.
        return "string";
    }
```